compiler/src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -187,7 +187,8 @@ class SyntheticMethods(thisTransformer: DenotTransformer) {
      */
     def valueHashCodeBody(implicit ctx: Context): Tree = {
       assert(accessors.length == 1)
-      ref(accessors.head).select(nme.hashCode_).ensureApplied
+      if (accessors.head.info.isPhantom) Literal(Constant(0))
```

Comment:
I wonder if we should actually disallow them. Still this code should not thow for case classes with phantoms.